File: new/models/MasyarakatModel.js
import { Sequelize } from "sequelize";
import db from "../config/database.js";
import Pengaduan from "../models/NgaduModel.js"

const {DataTypes} = Sequelize;

const Masyarakat = db.define('masyarakat',{
    nik : {
        type :DataTypes.INTEGER,
        autoIncrement : true,
        primaryKey : true
    },
    nama : DataTypes.STRING,
    username : DataTypes.STRING,
    password : DataTypes.STRING,
    telp : DataTypes.STRING
},{
    freezeTableName : true
});



export default Masyarakat;

// (async()=>{
//     await db.sync({});
// }) ();

File: lib/masyarakat/model/masyarakat_model.dart
class MasyarakatModel{
  int? nik;
  String? nama;
  String? username;
  String? password;
  String? telp;


  MasyarakatModel(this.nik, this.nama, this.username);

  MasyarakatModel.fromJson(Map<String, dynamic> json) {
    nik = json['nik'];
    nama = json['nama'];
    username = json['username'];
    password = json['password'];
    telp = json['telp'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['nik'] = this.nik;
    data['nama'] = this.nama;
    data['username'] = this.username;
    return data;
  }
}